Adapt the users service to the HTTP/JSON API (#1117)
openSUSE/agama: web/README.md

which does not contain any development tools, you can develop the web frontend
easily from your workstation.

Example of running from different machine:

```
  # backend machine
  # using ip of machine instead of localhost is important to be network accessible

>   agama-web-server serve --address 10.100.1.1:3030

Reviewer: ```suggestion
  agama-web-server serve --address 10.100.1.1:3000
```

Author: and that reminds me that I should update that documentation now with ssl enabled

  # frontend machine

>   # ESLINT=0 is useful during development when there is some styling issues

Reviewer: ```suggestion
  # ESLINT=0 is useful to ignore linter problems during development
```

  ESLINT=0 AGAMA_SERVER=10.100.1.1:3000 npm run server
```

### Debugging Hints

> There are more locations where to look when something does not work and require debugging.

Reviewer: ```suggestion
There are several places to look when something does not work and requires debugging.
```

> The first location to check when something does not work is browser console which can give

Reviewer: ```suggestion
The first place is the browser's console which can give
```

some hints. The second location to check for errors or warnings is output of `npm run server`

> where some issues when communicating with backend can be seen. And last but on least is

Reviewer: ```suggestion
where you can find issues when communicating with the backend. And last but on least is
```

> journal on backend machine where is logged backend activity `journalctl -b`.

Reviewer: Thanks a lot for documenting this. I would add: `busctl monitor --address /run/agama/bus` as another source of information.

Author: well, I think that this busctl monitor and other useful tools for dbus observing belongs more to backend documentation. Either service or rust dir. Here I try to focus mostly on debugging issues with web UI.

Reviewer: Yes, sure, but given that you mentioned `journalctl` too, I think it might be useful.

Author: well, I mention journalctl mainly because you see there if request from web UI lands to backend. If it is correctly send there, then I would continue with backend debugging. But maybe we can link backend debugging document from this one.

Reviewer: Well, I would avoid having different documents about debugging. All the information about that matter should be in a single place.

Author: ok, so lets add it.
